### PR TITLE
Implement IPv6 support for server and direct connect

### PR DESCRIPTION
**Pull request by pchote**
This pr is #17571 rebased on current bleed, with an extra commit added to solve my nits about duplication in the error messages. See the previous pr for discussion and history.

### CODE REVIEW

**Comment by pchote** [comment]
Adding my :+1: for the original changes.

File: OpenRA.Game/Network/OrderManager.cs [hunk 1]
```diff
@@ -28,11 +28,10 @@ public sealed class OrderManager : IDisposable
 		public Session.Client LocalClient { get { return LobbyInfo.ClientWithIndex(Connection.LocalClientId); } }
 		public World World;
 
-		public readonly string Host;
-		public readonly int Port;
+		public readonly ConnectionTarget Endpoint;
 		public readonly string Password = "";
 
-		public string ServerError = "Server is not responding";
+		public string ServerError = null;
```
Comment (sorcerer86pt) [inline OpenRA.Game/Network/OrderManager.cs:34]: why the null here?

File: OpenRA.Game/Server/Server.cs [hunk 2]
```diff
@@ -129,15 +126,43 @@ public void EndGame()
...
-			listener.Start();
-			var localEndpoint = (IPEndPoint)listener.LocalEndpoint;
-			Ip = localEndpoint.Address;
-			Port = localEndpoint.Port;
+			SocketException lastException = null;
+			var checkReadServer = new List<Socket>();
+			foreach (var endpoint in endpoints)
+			{
+				var listener = new TcpListener(endpoint);
+				try
```
Comment (sorcerer86pt) [inline OpenRA.Game/Server/Server.cs:138]: why the nested double try catch code here

File: OpenRA.Game/Network/OrderManager.cs (hunk 1 above)
Comment (pchote) [reply OpenRA.Game/Network/OrderManager.cs:34]: This enables the connection error dialog to fall back to a message set on the connection object if the server does not specifically override it. If neither are set, the connection dialog will fall back to "Unknown error".

File: OpenRA.Game/Server/Server.cs (hunk 2 above)
Comment (pchote) [reply OpenRA.Game/Server/Server.cs:138]: The inner try/catch is dealing with systems that lack dual stack support - we expect exceptions there, and can safely ignore them. The outer try/catch is for unexpected exceptions which can't be ignored.

File: OpenRA.Game/Network/OrderManager.cs (hunk 1 above)
Comment (sorcerer86pt) [reply OpenRA.Game/Network/OrderManager.cs:34]: thanks for the quick response.

File: OpenRA.Game/Server/Server.cs (hunk 2 above)
Comment (sorcerer86pt) [reply OpenRA.Game/Server/Server.cs:138]: Can you add this explanation as a comment to the code. Someone, at sometime will have the same question, and a comment here would explain this.

**Review by sorcerer86pt — APPROVED** [review]

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/51d721eb331240130dee805cdbb95fb0761af49a...bfb94c55a1521fd5411cd634b855417a949aab28

**Pull request merged**